Improve Code Coverage in YearlyEventCalender.tsx

SukhvirKooner opened the pull request:
<!--
This section can be deleted after reading.

We employ the following branching strategy to simplify the development process and to ensure that only stable code is pushed to the `master` branch:

- `develop`: For unstable code: New features and bug fixes.
- `master`: Where the stable production ready code lies. Only security related bugs.

NOTE!!!

ONLY SUBMIT PRS AGAINST OUR `DEVELOP` BRANCH. THE DEFAULT IS `MAIN`, SO YOU WILL HAVE TO MODIFY THIS BEFORE SUBMITTING YOUR PR FOR REVIEW. PRS MADE AGAINST `MAIN` WILL BE CLOSED.

-->

<!--
Thanks for submitting a pull request! Please provide enough information so that others can review your pull request.
-->

**What kind of change does this PR introduce?**
 Adding Tests
<!-- E.g. a bugfix, feature, refactoring, etc… -->

**Issue Number:**

Fixes #3047 <!--Add related issue number here.-->

**Snapshots/Videos:**
<img width="853" alt="Screenshot 2025-02-03 at 12 33 51 AM" src="https://github.com/user-attachments/assets/eb9e2b9f-b2d9-4670-897a-6e7f0e86e4f2" />


<!--Add snapshots or videos wherever possible.-->

**If relevant, did you update the documentation?**

<!--Add link to Talawa-Docs.-->

**Summary**

- This PR adds comprehensive unit tests for the `YearlyEventCalender `component to improve test coverage and ensure robust behavior. 
-  The Test don't simulate user interactions that set `selectedDate` equal to the rendered date. That's why `/* istanbul ignore next */` is added to line 229—to prevent unnecessary coverage drops for production-only behavior.
- `/* istanbul ignore next */` was removed from line 143 because The tests cover those lines 

<!-- Explain the **motivation** for making this change. What existing problem does the pull request solve? -->
<!-- Try to link to an open issue for more information. -->

**Does this PR introduce a breaking change?**
No
<!-- If this PR introduces a breaking change, please describe the impact and a migration path for existing applications. -->

## Checklist

### CodeRabbit AI Review
- [ ] I have reviewed and addressed all critical issues flagged by CodeRabbit AI
- [ ] I have implemented or provided justification for each non-critical suggestion
- [ ] I have documented my reasoning in the PR comments where CodeRabbit AI suggestions were not implemented

### Test Coverage
- [ ] I have written tests for all new changes/features
- [ ] I have verified that test coverage meets or exceeds 95%
- [ ] I have run the test suite locally and all tests pass


**Other information**

<!--Add extra information about this PR here-->

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

<!--Yes or No-->

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **Tests**
	- Enhanced test coverage for the yearly calendar to ensure events display accurately based on user roles and various interaction scenarios.
- **Chores**
	- Updated internal test configurations to improve quality assurance and reliability of event filtering and calendar rendering.
- **Bug Fixes**
	- Improved handling of event filtering logic for different user roles, ensuring correct event visibility.
- **Documentation**
	- Streamlined documentation for mock variables to reflect current data structures and types used in the application.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

SukhvirKooner commented on src/components/EventCalendar/YearlyEventCalender.tsx:
```diff
@@ -227,6 +226,7 @@ const Calendar: React.FC<InterfaceCalendarProps> = ({
             ? styles.day__today
             : '',
           date.getMonth() !== monthInx ? styles.day__outside : '',
+          /* istanbul ignore next */
```
Yes, The Test don't simulate user interactions that set selectedDate equal to the rendered date. That's why `/* istanbul ignore next */` is added to line 229—to prevent unnecessary coverage drops for production-only behavior.

palisadoes reviewed (changes requested):
1. Please remove the Istanbul ignore statements. 
1. They give a false sense of high code quality

Aim to get as close to 100% coverage without them

SukhvirKooner commented:
Hi @palisadoes  
Removed `/* istanbul ignore next */` statements and optimized test coverage to be as close to 100% as possible.
<img width="852" alt="Screenshot 2025-02-10 at 11 00 49 PM" src="https://github.com/user-attachments/assets/b8fcc1c7-d9fa-4583-b9e3-f0047d6018e7" />

palisadoes commented:
1. Please fix the failing tests
2. Make sure code rabbit.ai approves your changes
3.

SukhvirKooner commented:
Hi @palisadoes , 
**I've completed all the requested changes:**
1. Fixed the failing tests.
2. Got approval from code rabbit.ai.
3. Updated the PR title to be more searchable.

Additionally, I noticed that some files were modified automatically. Should I revert those changes, or are they expected?

SukhvirKooner commented:
Hey,
I was a bit occupied over the past few days, but I'm back now and will continue working on this.

noman2002 commented:
@SukhvirKooner closing this PR, submit a new one with relevant changes.
